tools/workspace/new_release.py

```diff
@@ -219,105 +229,118 @@ def _is_modified(repo, path):
+
+        message = f"[workspace] Upgrade {workspace_name}"
+        if _smells_like_a_git_commit(new_commit):
+            message += " to latest commit"
         else:
-            commit = True
-    else:
-        commit = False
-
-    # Figure out what to upgrade.
-    old_commit, new_commit = _handle_github(workspace_name, gh, data)
-    if old_commit == new_commit:
-        raise RuntimeError(f"No upgrade needed for {workspace_name}")
-    elif new_commit is None:
-        raise RuntimeError(f"Cannot auto-upgrade {workspace_name}")
-    print("Upgrading {} from {} to {}".format(
-        workspace_name, old_commit, new_commit))
-
-    # Locate the two hexadecimal lines we need to edit.
-    commit_line_re = re.compile(
-        r'(?<=    commit = ")(' + re.escape(old_commit) + r')(?=",)')
-    checksum_line_re = re.compile(
-        r'(?<=    sha256 = ")([0-9a-f]{64})(?=",)')
-    commit_line_num = None
-    checksum_line_num = None
-    for i, line in enumerate(lines):
-        match = commit_line_re.search(line)
-        if match:
-            assert commit_line_num is None
-            commit_line_num = i
-            commit_line_match = match
-        match = checksum_line_re.search(line)
-        if match:
-            assert checksum_line_num is None
-            checksum_line_num = i
-            checksum_line_match = match
-    assert commit_line_num is not None
-    assert checksum_line_num is not None
-
-    # Download the new source archive.
-    print("Downloading new archive...")
-    new_url = f"https://github.com/{repository}/archive/{new_commit}.tar.gz"
-    hasher = hashlib.sha256()
-    with open(f"{temp_dir}/{new_commit}.tar.gz", "wb") as temp:
-        with urllib.request.urlopen(new_url) as response:
-            while True:
-                data = response.read(4096)
-                if not data:
-                    break
-                hasher.update(data)
-                temp.write(data)
-            new_checksum = hasher.hexdigest()
-
-    # Update the repository.bzl contents and then write it out.
-    lines[commit_line_num] = commit_line_re.sub(
-        new_commit, lines[commit_line_num])
-    lines[checksum_line_num] = checksum_line_re.sub(
-        new_checksum, lines[checksum_line_num])
-    with open(bzl_filename + ".new", "w", encoding="utf-8") as f:
-        for line in lines:
-            f.write(line)
-    os.rename(bzl_filename + ".new", bzl_filename)
-
-    # Copy the downloaded tarball into the repository cache.
-    print("Populating repository cache ...")
-    subprocess.check_call(["bazel", "fetch", "//...", f"--distdir={temp_dir}"])
-
-    message = f"[workspace] Upgrade {workspace_name}"
-    if _smells_like_a_git_commit(new_commit):
-        message += " to latest commit"
-    else:
-        message += f" to latest release {new_commit}"
+            message += f" to latest release {new_commit}"
+        messages.append(message)
```

Comment:
I don't think this works; it's going to result in a mangled commit message. I think if there is more than one repository, the message [summary] should be "[workspace] Upgrade foo, bar to latest". More specific information can be in the commit body, if needed.